test/LaunchDarkly.EventSource.Tests/EventSourceTests.cs

```diff
@@ -377,21 +374,19 @@ public async Task Given_204_when_the_http_response_is_received_then_error_event_
 
             var evt = new EventSource(new Configuration(_uri, handler));
 
+            var receiver = new ErrorReceiver(evt);
+
             //Act
-            var raisedEvent = await Assert.RaisesAsync<ExceptionEventArgs>(
-                h => evt.Error += h,
-                h => evt.Error -= h,
-                () => evt.StartAsync());
+            await evt.StartAsync();
```

Comment:
This is an example of the unit tests verifying the wrong behavior. `await evt.StartAsync()` will only return once the stream has been completely shut down. An HTTP error is not supposed to shut it down - it should reconnect. The test only worked because the stream was broken.

In the new version, `ErrorReceiver` takes care of closing the stream once it has received an error.